spec/rubocop/cop/performance/deterministic_regexp_spec.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Performance::DeterministicRegexp do
+  subject(:cop) { described_class.new }
+
+  it 'accepts methods other than split' do
+    expect_no_offenses("'a,b,c'.insert(2, 'a')")
+  end
+
+  it 'accepts when split does not receive a regexp' do
+    expect_no_offenses("'a,b,c'.split(',')")
+  end
+
+  it 'accepts when split does not receive a deterministic regexp' do
+    expect_no_offenses("'a,b,c'.split(/,+/)")
+  end
+
+  it 'registers an offense when the method is split' do
+    expect_offense(<<~RUBY)
+      'a,b,c'.split(/,/)
+      ^^^^^^^^^^^^^^^^^^ Use string as argument instead of regexp.
+    RUBY
+  end
+
+  context 'when auto-correcting' do
+    describe 'converts regexp to string' do
+      it 'corrects when the length of the pattern and replacement are one' do
+        new_source = autocorrect_source("'a,b,c'.split(/,/)")
+
+        expect(new_source).to eq("'a,b,c'.split(',')")
+      end
+    end
+  end
```

Comment:
Can you use `expect_correction` API for auto-correction testing? I wrote an example in the comment above.